### ISSUE TITLE
`Transport`: many options in `SshTransport` seems are no longer in use.

### ISSUE THREAD

**Issue by khsrali** [aiidateam/aiida-core#6523]
Maybe I'm wrong, but it seems these options are taken from users for no apparent reason:

`look_for_keys`
`allow_agent`
`compress`
`gss_auth`
`gss_kex`
`gss_deleg_creds`
`gss_host`
`timeout` (wrong name in use `connecttimeout`, this should be a bug)

**Comment by sphuber** [comment]
I am pretty sure that they _are_ used, namely when opening the connection through the paramiko client:
https://github.com/aiidateam/aiida-core/blob/a5da4eda131f844c3639bdb01a256b9e9a7873a2/src/aiida/transports/plugins/ssh.py#L498

**Comment by khsrali** [comment]
Sure, thanks!
However it seems here instead of `timeout` is looking for `connecttimeout`
```python
@classmethod
def _get_timeout_suggestion_string(cls, computer):
    """Return a suggestion for the specific field.

    Provide 60s as a default timeout for connections.
    """
    config = parse_sshconfig(computer.hostname)
    return str(config.get('connecttimeout', '60'))
```

**Comment by sphuber** [comment]
This method is just providing a default value based on the SSH config file though. And there apparently the relevant attribute is called `connecttimeout`. It doesn't necessarily have to have the same name that is then defined by the paramiko client.

**Comment by khsrali** [comment]
Yes I see now, `connecttimeout` is being parsed from `~/.ssh/config` by `paramiko` I just don't see where in `paramiko`  explicitly

**Comment by sphuber** [comment]
Not sure which part is not clear. The [`ConnectTimeout` is a parameter for SSH](https://man7.org/linux/man-pages/man5/ssh_config.5.html). The [transport uses `paramiko.SSHConfig` to parse the SSH config file and returns an object that allows to address settings in lower-case version](https://docs.paramiko.org/en/latest/api/config.html#paramiko.config.SSHConfig.lookup), i.e., for `ConnectionTimeout` it would be `config['connecttimeout']`. But the [`paramiko.SSHClient.connect` method uses `timeout`](https://docs.paramiko.org/en/2.4/api/client.html#paramiko.client.SSHClient.connect) for the connection timeout.

**Comment by khsrali** [comment]
Thank you for the clarification, it already became clear.
